### PR TITLE
feat: add synchronization workflow

### PR DESCRIPTION
**Pull request by asterwyx**
Add synchronization workflow which will be used in DTK components to
synchronize sources from DTK5 to DTK6 repository. It's a reusable
workflow.

Log: add reusable synchronization workflow

### CODE REVIEW

File: .github/workflows/synchronize-to-dtk6.yml [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+          ls $(pwd)
+          cd ${{ github.workspace }}/source
+          version=$(git describe)
+          cd ${{ github.workspace }}/dest
+          git checkout -b synchronize-source
+          echo "GH_TOKEN=${{ steps.get-token.outputs.result }}" >> $GITHUB_ENV
+          echo "GH_TOKEN: ${GH_TOKEN}"
+          rsync -avzn --delete --exclude=.git --exclude=.obs --exclude=debian --exclude=archlinux --exclude-from=.syncexclude ${{ github.workspace }}/source ${{ github.workspace }}/source
+          git add :/
+          git commit -m "sync: from dtkcore" -m "Synchronize source files from dtkcore ${version}."
```
Comment (tsic404) [inline .github/workflows/synchronize-to-dtk6.yml]: sync from: dtkcore  not ${{ inputs.dest_repo }} ?

File: .github/workflows/synchronize-to-dtk6.yml [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+          ls ${{ github.workspace }}/source
+          ls ${{ github.workspace }}/dest
+          ls $(pwd)
+          cd ${{ github.workspace }}/source
+          version=$(git describe)
+          cd ${{ github.workspace }}/dest
+          git checkout -b synchronize-source
+          echo "GH_TOKEN=${{ steps.get-token.outputs.result }}" >> $GITHUB_ENV
+          echo "GH_TOKEN: ${GH_TOKEN}"
+          rsync -avzn --delete --exclude=.git --exclude=.obs --exclude=debian --exclude=archlinux --exclude-from=.syncexclude ${{ github.workspace }}/source ${{ github.workspace }}/source
```
Comment (tsic404) [inline .github/workflows/synchronize-to-dtk6.yml]:
```suggestion
          rsync -avzn --delete --exclude=.git --exclude=.obs --exclude=debian --exclude=archlinux --exclude-from=.syncexclude ${{ github.workspace }}/source ${{ github.workspace }}/dest
```

Comment (tsic404) [inline .github/workflows/synchronize-to-dtk6.yml]: .syncexclude exists?

**Review by hudeng-go — APPROVED** [review]

**Review by tsic404 — APPROVED** [review]

**Review by 18202781743 — APPROVED** [review]

**Pull request merged**